Realm/RLMUUID.mm

```diff
@@ -0,0 +1,45 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2020 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#import "RLMUUID_Private.hpp"
+#import <realm/uuid.hpp>
+
+@implementation NSUUID (RLMUUIDSupport)
+
+- (instancetype)initWithRealmUUID:(realm::UUID)rUuid {
+    self = [self initWithUUIDBytes: rUuid.to_bytes().data()];
+    return self;
+}
+
+- (realm::UUID)uuidValue {
+    return realm::UUID([self.UUIDString cStringUsingEncoding: NSUTF8StringEncoding]);
+}
+
+- (NSComparisonResult)compare:(NSUUID *)other {
```

Comment:
Defining compare: on a standard library type isn't really something we should do as a library.

Reply:
Any alternative?

Reply:
I think the idea is that we simply remove it, since it's adding base functionality that has nothing to do with Realm.

Reply:
I need this call for tests so I'll implement it as a category available in tests target